src/blocks/button/Button.tsx

```diff
@@ -46,7 +49,15 @@ const StyledButton = styled.button<ButtonProps>`
   }
 
   /* Button variant CSS styles */
-  ${({ variant }) => getButtonVariantStyles(variant || 'primary')}
+  ${({ variant, loading }) => getButtonVariantStyles(variant || 'primary', loading || false)}
+
+  ${({ iconOnly }) =>
+    !iconOnly &&
+    `[role='spinner'] {
+        margin-right: var(--spacing-xxxs);
+      };`}
```

Comment:
Not the correct approach. Button tag should provide the gap in the child elements.